modules/OCPP201/OCPP201.cpp

```diff
@@ -943,7 +939,35 @@ void OCPP201::ready() {
 
         evse->subscribe_ev_info([this, evse_id](const types::evse_manager::EVInfo& ev_info) {
             if (ev_info.soc.has_value()) {
-                this->evse_soc_map[evse_id] = ev_info.soc.value();
+                this->evse_soc_map[evse_id] = ev_info.soc;
+            }
+            if (ev_info.evcc_id.has_value()) {
+                // TODO(mlitre): Update ConnectedEVVehicleId DeviceModel
+                auto tx_data = this->transaction_handler->get_transaction_data(evse_id);
+                if (tx_data->id_token.has_value()) {
+                    auto info_vector = tx_data->id_token->additionalInfo.has_value()
+                                           ? tx_data->id_token->additionalInfo.value()
+                                           : std::vector<ocpp::v2::AdditionalInfo>{};
+                    if (!tx_data->id_token->additionalInfo.has_value() or
+                        (tx_data->id_token->additionalInfo.has_value() and
+                         std::find_if(tx_data->id_token->additionalInfo->cbegin(),
+                                      tx_data->id_token->additionalInfo->cend(),
+                                      [ev_info](const ocpp::v2::AdditionalInfo& info) {
+                                          return info.additionalIdToken.get() == ev_info.evcc_id.value();
+                                      }) == tx_data->id_token->additionalInfo->cend())) {
+                        ocpp::v2::AdditionalInfo info;
+                        info.additionalIdToken = ev_info.evcc_id.value();
+                        info.type = "EVCCID";
+                        info_vector.push_back(info);
+                        tx_data->id_token->additionalInfo = info_vector;
+                    }
+                } else {
+                    ocpp::v2::IdToken token;
+                    token.additionalInfo = {};
+                    token.additionalInfo->push_back(
+                        ocpp::v2::AdditionalInfo{ev_info.evcc_id.value(), "EVCCID", std::nullopt});
+                    tx_data->id_token = token;
```

Comment:
This could go into a separate member function, e.g. `void update_evcc_id_token(const std::string& evse_id, const std::string& evcc_id)`